lib/eex/test/eex/smart_engine_test.exs

```diff
@@ -29,16 +29,25 @@ defmodule EEx.SmartEngineTest do
     assert_eval("1\n2\n3\n", "<%= for x <- [1, 2, 3] do %><%= x %>\n<% end %>")
   end
 
-  test "preserves line numbers" do
-    result = EEx.compile_string("<%= @hello %>", engine: EEx.SmartEngine)
-
-    Macro.prewalk(result, fn
-      {_left, meta, _right} ->
-        assert Keyword.get(meta, :line, 0) in [0, 1]
-
-      _ ->
-        :ok
-    end)
+  test "preserves line numbers in assignments" do
+    result = EEx.compile_string("foo\n<%= @hello %>", engine: EEx.SmartEngine)
+
+    found =
+      try do
+        Macro.prewalk(result, fn
+          {_left, meta, [_, :hello]} ->
+            assert Keyword.get(meta, :line) == 2
+            throw :found
```

Comment:
instead of throw, you can send a message to yourself and then assert it was received: `send self(), :found` and then `assert_received :found`.